.github/workflows/pull-request.yaml

```diff
@@ -6,32 +6,38 @@ on:
   push:
     branches: [main]
 jobs:
-  rust:
+  rust-misc:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - uses: Swatinem/rust-cache@v1
-      - run: |
-          npm install ganache-cli
-          node_modules/.bin/ganache-cli --networkId 5777 --gasLimit 10000000 --gasPrice 0&
       - run: |
           rustup --version
           rustup show
           cargo --version
           cargo fmt --version
           cargo clippy --version
       - run: cargo fmt --all -- --check
-      - run: cargo run --bin deploy --features bin
-        working-directory: contracts
       - run: cargo clippy --locked --workspace --all-features --all-targets -- -D warnings
-      - run: cargo test --locked --workspace --all-features
+  rust-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: Swatinem/rust-cache@v1
```

Comment:
I would have thought that having to download the cache twice would take longer than the time it takes to run clippy itself on a warm build but it looks like overall the CI time is down from 3m 37s [here](https://github.com/gnosis/gp-v2-services/actions/runs/483087532) to 2:42 on this PR.

But maybe something to keep in mind as our dependencies and thus the cache grows.

Reply:
Jobs run in parallel and have a unique cache (i think). So it's not going to be slower.

Reply:
Right, since we don't own/pay for the infrastructure we don't care about the global efficiency. My point was more that if two tasks have a large combined setup cost (downloading cache + build) and one runs its independent action in a few seconds it might not be optimal to run them in parallel and incur twice the setup cost globally.

Reply:
Agree. In this case the shared setup cost is only downloading the dependencies from crates io but not any compilation because of the different compilation targets, so it's quite low.